scan/src/App.re

```diff
@@ -53,7 +53,17 @@ let make = () => {
         </Row>
         <SearchBar />
       </div>
-      <HomePage />
+      /* route handle */
+      {switch (ReasonReactRouter.useUrl() |> Route.fromUrl) {
+       | HomePage => <HomePage />
+       | ScriptHomePage => <ScriptHomePage />
+       | ScriptIndexPage(codeHash, hashtag) => <ScriptIndexPage codeHash hashtag />
+       | TxHomePage => <TxHomePage />
+       | TxIndexPage(txHash, hashtag) => <TxIndexPage txHash hashtag />
+       | BlockHomePage => <BlockHomePage />
+       | BlockIndexPage(height, hashtag) => <BlockIndexPage height hashtag />
+       | _ => <NotFound />
```

Comment:
For NotFound, let's redirect to HomePage? Do it similar to what we did in `staking-reason`. We can change it to its own page later if we ever have design for 404 page.